internal/report/store.go

```diff
@@ -0,0 +1,121 @@
+package report
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+)
+
+// PolicyReportStore caches the latest version of PolicyReports
+type policyReportStore struct {
+	// store is a map of namespaces and namespaced PolicyReports
+	store map[string]RInterface
+	// clusterPR is the sole ClusterPolicyReport
+	clusterPR ClusterPolicyReport
+
+	rwm *sync.RWMutex
+}
+
+type PolicyReportStore interface {
+	// Add a PolicyReport to the Store
+	Add(r RInterface) error
+	// Get PolicyReport by namespace
+	Get(ns string) (RInterface, error)
+	// Get the ClusterPolicyReport
+	GetClusterWide() (RInterface, error)
+	// Update ClusterPolicyReport or PolicyReport. ns argument is used in case
+	// of namespaced PolicyReport
+	Update(r RInterface) error
+	// Delete PolicyReport by namespace
+	Remove(ns string) error
+	// Delete all namespaced PolicyReports
+	RemoveAllNamespaced() error
+	// Marshal the contents of the store into a JSON string
+	ToJSON() (string, error)
+}
+
+// NewPolicyReportStore construct a PolicyReportStore, initializing the
+// clusterwide ClusterPolicyReport
+func NewPolicyReportStore() PolicyReportStore {
+	return &policyReportStore{
+		store:     make(map[string]RInterface),
+		clusterPR: *NewCPR("clusterwide"),
+		rwm:       new(sync.RWMutex),
+	}
+}
+
+func (s *policyReportStore) Add(report RInterface) error {
+	s.rwm.Lock()
+	defer s.rwm.Unlock()
+	if report.GetType() == ClusterPolicyReportType {
+		cpr, _ := report.(*ClusterPolicyReport)
+		s.clusterPR = *cpr
+	} else {
+		s.store[report.GetNamespace()] = report
+	}
+	return nil
+}
+
+func (s *policyReportStore) Get(namespace string) (RInterface, error) {
```

Comment:
Rename that to `GetPolicyReport`, return a `PolicyReport`

Reply:
Done in https://github.com/kubewarden/audit-scanner/pull/40/commits/0329296a7ee8c77e295d16a18d64d32492974eee.